data_structures/static_rtree.hpp

```diff
@@ -642,10 +642,55 @@ class StaticRTree
         return result_coordinate.is_valid();
     }
 
+    /**
+     * Checks whether A is between B-range and B+range, all modulo 360
+     * e.g. A = 5, B = 5, range = 10 == true
+     *      A = -6, B = 5, range = 10 == false
+     *      A = -2, B = 355, range = 10 == true
+     *      A = 6, B = 355, range = 10 == false
+     *      A = 355, B = -2, range = 10 == true
+     *
+     * @param A the bearing to check, in degrees, 0-359, 0=north
+     * @param B the bearing to check against, in degrees, 0-359, 0=north
+     * @param range the number of degrees either side of B that A will still match
+     * @return true if B-range <= A <= B+range, modulo 360
+     * */
+    static bool IsBearingWithinBounds(const Angle A,
+                                      const Angle B,
+                                      const int range)
+    {
+
+        if (range >= 180) return true;
+        if (range <= 0) return false;
+
+        // Map both bearings into positive modulo 360 space
+        const int normalized_B = (int(B) < 0)?(int(B) % 360)+360:(int(B) % 360);
+        const int normalized_A = (int(A) < 0)?(int(A) % 360)+360:(int(A) % 360);
```

Comment:
Please no C-style casts, a la `(int)`. Those will try all casts, with even falling back to `reinterpret_cast` (which is bad). Do a `static_cast` here, maybe cache its value.

Or better: strongly typedef-ing does not prevent you from extending the type.
You could still add functions, such as:

    inline int normalize(const Angle& a) { return /* my normalization code */; }

and then reuse that throughout the codebase. You can even put your typedef and those functions into their own namespace. With ADL you only have to say `normalize(myns::Angle(90))`, as normalize is being searched in the `msns` namespace automatically (this is ADL).